### PR TITLE
Directory definitions corrected and made to work for federated.

### PR DESCRIPTION
**Pull request by edwardalee**
This PR makes sure that `LF_SOURCE_GEN_DIRECTORY` and `LF_TARGET_FILES_DIRECTORY` are properly defined using CMake.  The latter in particular is needed for user code to be able to access files that are copied using the `files` target directive. This works for both unfederated and federated, even for remote federates.

### CODE REVIEW

File: core/src/main/java/org/lflang/federated/launcher/FedLauncherGenerator.java [hunk 1]
```diff
@@ -124,7 +124,7 @@ public void doGenerate(List<FederateInstance> federates, RtiConfig rtiConfig) {
       target = user + "@" + host;
     }
 
-    shCode.append("#### Host is ").append(host);
+    shCode.append("#### Host is ").append(host).append("\n");
```
Comment (lhstrh) [inline core/src/main/java/org/lflang/federated/launcher/FedLauncherGenerator.java:127]: " " followed by newline looks strange and was probably not the intent?

File: core/src/main/resources/lib/c/reactor-c
Comment (lhstrh) [inline core/src/main/resources/lib/c/reactor-c:1]: Are there updates in `reactor-c` that are required by this PR?

File: core/src/main/java/org/lflang/federated/launcher/FedLauncherGenerator.java (hunk 1 above)
Comment (edwardalee) [reply core/src/main/java/org/lflang/federated/launcher/FedLauncherGenerator.java:127]: ??? It's `host` followed by newline.  Previously the code generator was producing:
```
#### Host is fooecho something or other
```

File: core/src/main/resources/lib/c/reactor-c
Comment (edwardalee) [reply core/src/main/resources/lib/c/reactor-c:1]: No, but reactor-c was (as usual) not aligned with main, and that's where we want it to be for the tests to mean something.

File: core/src/main/java/org/lflang/federated/launcher/FedLauncherGenerator.java (hunk 1 above)
Comment (lhstrh) [reply core/src/main/java/org/lflang/federated/launcher/FedLauncherGenerator.java:127]: You're right; I missed the first append.

**Review by lhstrh — APPROVED** [review]

**Pull request merged**